docs/contribution-guides/Development-Standards-Guide.md

```diff
@@ -226,3 +242,301 @@ to just leave it as a public field without a property. As always, make sure to d
+
+```cs
+// bad code.
+
+classA : MonoBehavior{
+    void DoX_InTransform();
+}
+
+classB : MonoBehavior{
+    void DoX_InTransform();
+}
+
+classC : MonoBehavior{
+    void DoX_InTransform();
+}
+```
+
+```cs
+// good code.
+public static class TransformExtensions
+{
+    //can be called anywhere when they access Transform.
+    public static void DoX(this Transform t, float value)
+    {
+        //Common task here.
+    }
+}
+```
+
+## 2. Performance gotchas!
+
+* Unity still uses Mono, which means we do not get the performance benefits of .NET 6 and 7. As a result, minimise the use of things like [LINQ](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/) unless you know what you're doing.
+
+* As a general rule of thumb, avoid doing any performance intensive tasks or calculations on the game server. Your code may only take up 0.5% of CPU time but when 100 systems do this, it becomes inefficent and a glaring issue that someone has to deal with in the future.
+
+* Avoid using the `UpdateManager` whenever possible and bind yourself to an event driven archticture/design for your code. Not only is it good for code quality; But it's also good for performance when your code is designed in a way that expects functions to be called every once and a while rather than every tick/frame.
+
+* `GetComponent<>`, `GetComponentInChildren()` and `TryGetComponent<>()` are largely inefficent and can be incredibly slow. Use `CommonComponents` whenever possible or cache references to components.
+```cs
+//How to cache references.
+[SerializeField] private SpriteHandler itemSprite;
+
+void Awake()
+{
+    if(itemSprite == null) itemSprite = GetComponentInChildren<SpriteHandler>();
+}
+```
+
+* [Unity Coroutines](https://docs.unity3d.com/Manual/Coroutines.html) are great for moving tasks that require things to happen over time from the `Update()` loop into an asynchronous workflow that allows you to spread, pause and kill tasks freely over multiple frames. However, do be aware that it's a double edged sword as `IEnumerator` does allocate memory whenever `StartCoroutine` invokes a coroutine and you can only invoke them from inside Monobehaviors. Use them wisely.
+
+_TLDR: Coroutines trades CPU performance for memory_
+
+_"It’s best practice to condense a series of operations down to the fewest number of individual coroutines possible. Nested coroutines are useful for code clarity and maintenance, but they impose a higher memory overhead because the coroutine tracks objects." - Unity Docs._
+
+* Use Time-Splitting whenever your iterators/loops are attempting to go through a large amount of objects/tasks.
+```cs
+//How to time split.
+
+var currentIndex = 0;
+var maximumIndexesPerFrame = 20;
+
+foreach(var item in LargeList)
+{
+    if(currentIndex >= maximumIndexesPerFrame)
```

Comment:
We have an extension for enumerables to work in chunks

Reply:
what is it called?

Reply:
I will look for it after lunch

Reply:
https://github.com/unitystation/unitystation/blob/d5679c962287120d80dbe627452903d78ade4628/UnityProject/Assets/Scripts/Util/SweetExtensions.cs#L389

Reply:
I don't think it awaits a frame like I've written here.